cli/config/makeBabelConfig.js

```diff
@@ -0,0 +1,74 @@
+const browserTargets = require('./.browserlistrc')
+const jestTargets = { node: 'current' }
+
+const getBabelModuleType = moduleType => {
+    switch (moduleType) {
+        case 'cjs':
+        case 'commonjs':
+            return 'commonjs'
+        case 'systemjs':
+        case 'umd':
+        case 'amd':
+            return moduleType
+        case 'es':
+        default:
+            return false
+    }
+}
+const makeBabelConfig = ({ moduleType, mode }) => {
+    const isTest = mode === 'test'
+
+    return {
+        presets: [
+            require('@babel/preset-react'),
+            require('@babel/preset-typescript'),
+            [
+                require('@babel/preset-env'),
+                {
+                    modules: isTest
+                        ? 'commonjs'
+                        : getBabelModuleType(moduleType),
+                    targets: isTest ? jestTargets : browserTargets,
+                },
+            ],
+        ],
+        plugins: [
+            // Adds syntax support for import()
+            require('@babel/plugin-syntax-dynamic-import').default,
+
+            /*
+             * These plugins actually transform code
+             */
+
+            // Automatically include a React import when JSX is present
+            require('babel-plugin-react-require'),
```

Comment:
We might not need this thanks to the new JSX transform https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

Reply:
True!  However since we haven't bumped the required version of `react` yet (still `react@^16.8`) I think we need to keep it for now